app/download/views.py

```diff
@@ -13,7 +13,14 @@ def download_document(service_id, document_id):
         return jsonify(error='Missing decryption key'), 400
 
     try:
-        document = document_store.get(service_id, document_id, request.args.get('key'))
+        key = base64_to_bytes(request.args['key'])
+        if len(key) != 32:
```

Comment:
I wonder if this check should be part of `document_store.get`, since it's about the size of the S3 encryption key, not the view itself

Reply:
i might just remove the check entirely - AWS will throw an error (which we'll re-raise as a DocumentStoreError and handle below. I don't think there's enough of a meaningful difference between "key has wrong length" and "key has right length but wrong data" to bother differentiating them